src/Stratis.Bitcoin.Features.BlockStore/BlockStoreQueue.cs

```diff
@@ -289,7 +289,17 @@ public void AddToPending(ChainedHeaderBlock chainedHeaderBlock)
         {
             lock (this.blocksCacheLock)
             {
-                this.pendingBlocksCache.TryAdd(chainedHeaderBlock.ChainedHeader.HashBlock, chainedHeaderBlock);
+                bool added = this.pendingBlocksCache.TryAdd(chainedHeaderBlock.ChainedHeader.HashBlock, chainedHeaderBlock);
+
+                if (added)
+                {
+                    this.logger.LogTrace("Block '{0}' was added to pending.", chainedHeaderBlock.ChainedHeader);
+                }
+                else
+                {
+                    this.logger.LogCritical("Unable to add block '{0}' to pending!", chainedHeaderBlock.ChainedHeader);
+                    throw new Exception("Can't add block to pending!");
```

Comment:
What will happen to the node if this fails?

Reply:
node stops advancing

Reply:
Why do you suspect this will even happen?

Reply:
Because logs @maciejzaleski provided indications that https://github.com/stratisproject/StratisBitcoinFullNode/issues/2595 is caused by inability to get a block from CHT cache and BSQ. 
So need to investigate missing blocks and make sure that they were saved in BSQ correctly and then not deleted.  

Basically 2595 is about node trying to do a big reorg (100+ blocks) which leads to inability to get blocks for connection. 
This might be CM, CHT or BSQ issue.

Reply:
ok yes I am very much in agreement, by the way why did we use the number 100? and is it configurable?

Reply:
it's not configurable. 
value of 100 is pretty much random. 

for PoS networks reasonable value is between 10 and MaxReorg